### PR TITLE
Added MappingField

### PR DESCRIPTION
**Pull request by Anton-Shutik**

### CODE REVIEW

**Comment by auvipy** [comment]
This branch has conflicts that must be resolved

**Comment by auvipy** [comment]
@Anton-Shutik hi thanks for your PR, would you please review the contribution guide and fix the merge conflict?

**Comment by Anton-Shutik** [comment]
Yep, 

But will do a bit later, in day or two

Thanks

**Comment by auvipy** [comment]
sure 
:)

**Comment by Anton-Shutik** [comment]
@auvipy Done

File: rest_framework_extensions/fields.py [hunk 1]
```diff
@@ -25,4 +26,38 @@ class Meta:
...
+
+    def to_representation(self, value):
+        if value in self.mapping:
+            return self.mapping[value]
+        self.fail('key_not_found', value=value)
+
+    def to_internal_value(self, data):
+        if data in self.reverse_mapping:
+            return self.reverse_mapping[data]
+        self.fail('value_not_found', value=data)
```
Comment (auvipy) [inline rest_framework_extensions/fields.py]: no new line at the end (this error could be checked automaticaly by using flake8)

**Comment by auvipy** [comment]
+ you will need to provide functional and unit tests for the feature you requested. please review the readme section for contributing guideline.

**Comment by Anton-Shutik** [comment]
I don't see any tests for fields in the app like DRF has https://github.com/tomchristie/django-rest-framework/pull/3097/files

Can you add some base class for Field tests like in DRF ?

**Comment by auvipy** [comment]
well this package is dependent on DRF itself! so good idea is importing the base class from direct drf itself :dart:

**Comment by auvipy** [comment]
also please add some docs :D

**Comment by auvipy** [comment]
would you please squash the commits to a logical commit for new build? and rebase?

**Comment by auvipy** [comment]
or just resend a new PR for newer CI build

**Comment by auvipy** [comment]
closing just resend to rebuild against new CI settings

**Pull request closed without merging**